service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/user/shiro/UserCredentialsServiceImpl.java

```diff
@@ -90,4 +105,58 @@ public Credential changePasswordRequest(PasswordChangeRequest passwordChangeRequ
             return credentialService.update(passwordCredential);
         });
     }
+
+
+    @Override
+    public Credential resetPassword(KapuaId scopeId, KapuaId credentialId, PasswordResetRequest passwordResetRequest) throws KapuaException {
+        Credential credential = credentialService.find(scopeId, credentialId);
```

Comment:
It is better to do first:
- argument validation: to throw proper KapuaIllegalArgumentException
- checkAccess: to avoid guessing of entity existence base on the response time of the request